docs/source/example_notebooks/dowhy_efficient_backdoor_example.ipynb

```diff
@@ -0,0 +1,519 @@
+    "                        node[id \"pre-game prop\" label \"pre-game prop\"]\n",
+    "                        node[id \"intra-game prop\" label \"intra-game prop\"]                       \n",
+    "                        node[id \"neuromusc fatigue\" label \"neuromusc fatigue\"]\n",
+    "                        node[id \"warm-up\" label \"warm-up\"]\n",
+    "                        node[id \"previous injury\" label \"previous injury\"]\n",
+    "                        node[id \"contact sport\" label \"contact sport\"]\n",
+    "                        node[id \"genetics\" label \"genetics\"]\n",
+    "                        node[id \"injury\" label \"injury\"]\n",
+    "                        node[id \"tissue disorder\" label \"tissue disorder\"]\n",
+    "                        node[id \"tissue weakness\" label \"tissue weakness\"]\n",
+    "                        edge[source \"coach\" target \"team motivation\"]\n",
+    "                        edge[source \"coach\" target \"fitness\"]\n",
+    "                        edge[source \"fitness\" target \"pre-game prop\"]\n",
+    "                        edge[source \"fitness\" target \"neuromusc fatigue\"]\n",
+    "                        edge[source \"team motivation\" target \"warm-up\"]\n",
+    "                        edge[source \"team motivation\" target \"previous injury\"]\n",
+    "                        edge[source \"pre-game prop\" target \"warm-up\"]\n",
+    "                        edge[source \"warm-up\" target \"intra-game prop\"]\n",
+    "                        edge[source \"contact sport\" target \"previous injury\"]\n",
+    "                        edge[source \"contact sport\" target \"intra-game prop\"]\n",
+    "                        edge[source \"intra-game prop\" target \"injury\"]\n",
+    "                        edge[source \"genetics\" target \"fitness\"]\n",
+    "                        edge[source \"genetics\" target \"neuromusc fatigue\"]\n",
+    "                        edge[source \"genetics\" target \"tissue disorder\"]\n",
+    "                        edge[source \"tissue disorder\" target \"neuromusc fatigue\"]\n",
+    "                        edge[source \"tissue disorder\" target \"tissue weakness\"]\n",
+    "                        edge[source \"neuromusc fatigue\" target \"intra-game prop\"]\n",
+    "                        edge[source \"neuromusc fatigue\" target \"injury\"]\n",
+    "                        edge[source \"tissue weakness\" target \"injury\"]\n",
+    "                        ]\n",
+    "\"\"\"\n",
+    "observed_node_names=[\"coach\", \"team motivation\", \"fitness\", \"neuromusc fatigue\",\n",
+    "                    \"warm-up\", \"previous injury\", \"contact sport\", \"tissue disorder\", \"injury\"]\n",
+    "treatment_name = \"warm-up\"\n",
+    "outcome_name = \"injury\"\n",
+    "G = CausalGraph(graph=graph_str, treatment_name=treatment_name, outcome_name=outcome_name,\n",
```

Comment:
On the first point, sounds good to me. Does the CausalModel api work fine if no data is passed at init? If someone is designing an observational study, and needs to choose which variables to measure (the main use case we had in mind when working on these problems), she will typically not have collected any data yet.

On the second one, an issue that I see is that to actually show that a strategy to compute backdoors is better than another (disregarding theory) we would need to run a little simulation, a single application to one data set would not do it in my view. I'd be happy to do something like that, but would leave it for a second PR if that's ok with you.

Reply:
Going back to your first point, I checked the CausalModel api and it does not seem very natural to use it for the example shown in this notebook, mostly for the reason I mentioned earlier, that if you are planning an observational study, you will not have any data yet.

Reply:
That's a good point. I like it because it brings up an interesting usecase that identification can be useful as an end-goal, before collecting the data. So far, in the dowhy examples, we had been using identification as a part of the causal estimation process. 

Btw, we are anyways planning to move away from the CausalModel class in the new [API](https://github.com/py-why/dowhy/wiki/API-proposal-for-v1). Thanks for highlighting this usecase, I will add it to the API design doc.